ledger/acctupdates.go

```diff
@@ -1814,7 +1874,16 @@ func (au *accountUpdates) lookupWithRewards(rnd basics.Round, addr basics.Addres
 			// we don't technically need this, since it's already in the baseAccounts, however, writing this over
 			// would ensure that we promote this field.
 			au.baseAccounts.writePending(macct)
-			return macct.pad.AccountData, nil
+			if full && macct.pad.ExtendedAssetHolding.Count != 0 {
+				macct.pad.Assets, macct.pad.ExtendedAssetHolding, err = au.accountsq.loadHoldings(macct.pad.ExtendedAssetHolding, macct.round)
+				var mismatchRoundErr *MismatchingDatabaseRoundError
+				if errors.As(err, &mismatchRoundErr) {
+					// go to waiting
+					// note, au.accountsMu is still held
+					goto retryLocked
+				}
```

Comment:
MismatchingDatabaseRoundError might happen when `au.baseAccounts` has older data than DB if round advances between `au.baseAccounts` and accountext reading. Retrying should sufficient - on next iteration we either do not hit this branch or data becomes synchronized.

Reply:
I believe that if you just want to check if the error is a specific error, a type assertion would suffice. But using `errors.As` would not be incorrect.